execution/scan/literal_selector.go

```diff
@@ -39,8 +40,25 @@ func NewNumberLiteralSelector(pool *model.VectorPool, opts *query.Options, val f
 		currentStep: opts.Start.UnixMilli(),
 		val:         val,
 	}
+
+	op.t = &model.NoopTimingInformation{}
+	if opts.EnableAnalysis {
+		op.t = &model.TimingInformation{CPUTime: time.Duration(0)}
+	}
+
+	return op
 }
+func (o *numberLiteralSelector) Analyze() (*model.TimingInformation, []model.ObservableVectorOperator) {
+	if telemetry, ok := o.t.(*model.TimingInformation); ok {
+		_, _ = o.Next(context.Background())
+		return telemetry, nil
+	}
+	return nil, nil
 
+}
+func (o *numberLiteralSelector) AddCPUTimeTaken(t time.Duration) {
+	o.t.AddCPUTimeTaken(t)
```

Comment:
I would suggest just embedding `model.OperatorTelemetry` directly so that we wouldn't have to write these functions. It would simply be possible to call `AddCPUTimeTaken` directly: https://go.dev/play/p/sYvyjlMaZh0